Makes it possible to have custom color for hover_pressed on CheckButton and CheckBox

fossegutten opened the pull request:
My first commit to any open source project. Be nice 😄 
This removes the need to have a custom stylebox for hover_pressed, to have custom color or icons.
Fixes #34727

fossegutten committed:
Fixes custom styles for hover_pressed on CheckButton and CheckBox.

akien-mga commented:
> My first commit to any open source project. Be nice :smile: 

Congrats! :)

Code-wise it looks OK though I haven't reviewed in depth whether it does the right thing. I assigned two contributors who might be more familiar with this code and what would be expected in different scenarios.

The commit message should be amended to give more details about what the commit actually does, beyond just the issue number that it fixes. See https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#format-your-commit-logs-with-readability-in-mind for details, and http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#modifying-a-pull-request for instructions on editing a commit in a PR.

aaronfranke reviewed (changes requested):
Braces will be enforced in 4.0: https://github.com/godotengine/godot/issues/33027

At the very least, don't commit changes that specifically remove them (like line 96), but also you should probably add them everywhere here.

fossegutten commented:
Okay i didn't realize. I will wait for the reviews.
I discovered that adding a stylebox to the theme for hover_pressed does not work either. You need to put a stylebox on each node to make it work. Maybe there are some problems deeper in the code, not just the bracket issues i fixed.

fossegutten commented:
Managed to fix the stylebox issue. Using a theme now works :) Hope i updated my commit correctly.  👍

akien-mga commented:
Sorry for conflicting instructions :/ As @aaronfranke, we're going to enforce brackets around single line `if`/`else` blocks in 4.0, but this will be done all at once using a dedicated tool.

Until then, it's better to just use such brackets around new code, but not to modify existing code to follow that convention unless it's only a few lines or directly around the code being changed. So the code style fixes you made now make it somewhat harder to see what are the actual logic changes in this relatively sensitive code (`base_button.cpp` is the base for all button nodes, so any wrong change here will impact all Godot GUIs).

So while it's not bad per se now, I would prefer it if the PR did not change the style of code which does not have logic changes, so that we can easily review each changed line.

You'd also have to squash the commits together so that the PR contains only one commit. See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase) for details.
About the commit message, your second commit is indeed more verbose than the first one, but I'd argue it's a bit too verbose now. It's not a big deal either, and figuring out how to write "good" commit logs is a matter of experience so don't worry about it.

I'd suggest something like this for the final commit:
```
Allow custom color for hover_pressed on CheckButton and CheckBox

Fixes #34727.
```

As outlined in [CONTRIBUTING.md](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#format-your-commit-logs-with-readability-in-mind), the commit message is structured in two sections: the title, which should be a short, single line, and then the body of the message which can be as long as necessary to provide more details. Only the title will appear in `git shortlog`, changelogs or when reviewing commits on GitHub, unless you click the `...` button for details. Official recommendation is to keep the title below 50 chars, but in my experience that's very difficult to achieve while keeping things readable, so it's OK to go slightly over it as done in my example (but I still reduced "Makes it possible to have" to "Allow" to be more concise).

I know it's a lot of info and changes requested, so it may feel overwhelming. But don't worry, we're also here to help you hone your FOSS contribution skills both for Godot and other future projects you make work on :)

fossegutten commented:
Thanks for your detailed instructions. I know you must be busy with helping the more experienced contributors for the upcoming 3.2 release! I will take it into consideration next time i try to contribute.

I managed to "squash" the commit and improve the commit message. Not sure how to remove the brackets again in an easy way, without making a new branch and commit. But i also removed a redundant return statement in base_button.cpp and moved out some code that was repeated in all 5 possible cases of the switch statement in button.cpp. So I think it should be worth keeping the changes.
